mir_eval/util.py

```diff
@@ -589,13 +589,58 @@ def recurse(v):
             recurse(v)
 
 
-def match_events(ref, est, window):
+def _outer_distance_mod_n(ref, est, modulus=12):
+    """Compute the absolute outer distance modulo n.
+    Using this distance, d(11, 0) = 1 (modulo 12)
+
+    Parameters
+    ----------
+    ref : np.ndarray, shape=(n,)
+        Array of reference event times.
+    est : np.ndarray, shape=(m,)
+        Array of estimated event times.
+    modulus : int
+        The modulus.
+        12 by default for octave equivalence.
+
+    Returns
+    -------
+    outer_distance : np.ndarray, shape=(n, m)
+        The outer circular distance modulo n.
+
+    """
+    abs_diff = np.abs(np.subtract.outer(ref, est))
+    return np.minimum(abs_diff, modulus - abs_diff)
+
+
+def _outer_distance(ref, est):
+    """Compute the absolute outer distance.
+    Computes |ref[i] - est[j]| for each i and j.
+
+    Parameters
+    ----------
+    ref : np.ndarray, shape=(n,)
+        Array of reference event times.
+    est : np.ndarray, shape=(m,)
+        Array of estimated event times.
+
+    Returns
+    -------
+    outer_distance : np.ndarray, shape=(n, m)
+        The outer 1d-euclidean distance.
+
+    """
+    return np.abs(np.subtract.outer(ref, est))
+
+
+def match_events(ref, est, window, outer_distance=_outer_distance):
```

Comment:
It's generally bad form to have objects (eg functions) as default arguments, as things can get ugly in case of mutation.

The (only slightly unsavory) fix for this is to have a default argument of `None` and then override it with custom logic in the function.

This also makes the docstrings prettier, since you get `outer_distance=None` instead of `outer_distance=<function at 0x324235rfdfsad45>`.

Also, I'd recommend renaming `outer_distance` to `distance`.